templates/aviation-checklist/pdf_aviation_checklist.go

```diff
@@ -0,0 +1,249 @@
+	// If you need a key, you can sign up and create a free one at https://cloud.unidoc.io.
+	err := license.SetMeteredKey(os.Getenv(`UNIDOC_LICENSE_API_KEY`))
+	if err != nil {
+		panic(err)
+	}
+
+	common.SetLogger(common.NewConsoleLogger(common.LogLevelDebug))
+}
+
+func main() {
+	c := creator.New()
+	c.SetPageMargins(30, 30, 92, 50)
+
+	// Read main content template.
+	mainTpl, err := readTemplate("templates/main.tpl")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read checklist JSON data.
+	checks, err := readChecklistData("checklist.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Draw main content template.
+	tplOpts := &creator.TemplateOptions{
+		HelperFuncMap: template.FuncMap{
+			"calcTableHeight": func(check *Checks) float64 {
+				stdFont := model.StdFontName(model.HelveticaName)
+				stdFontBold := model.StdFontName(model.HelveticaBoldName)
+
+				font, err := model.NewStandard14Font(stdFont)
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				fontBold, err := model.NewStandard14Font(stdFontBold)
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				titleSp := c.NewStyledParagraph()
+				chunk := titleSp.SetText(check.Title)
+				chunk.Style.Font = font
+
+				height := titleSp.Height()
+
+				for _, v := range check.Items {
+					var chunk *creator.TextChunk
+					itemSp := c.NewStyledParagraph()
+
+					if v.Action != nil && *v.Action != "" {
+						chunk = itemSp.SetText(*v.Action)
+						chunk.Style.Font = fontBold
+					} else {
+						label := ""
+						if v.Check != nil && *v.Check != "" {
+							label = *v.Check
+						}
+
+						val := ""
+						if v.Value != nil && *v.Value != "" {
+							label = *v.Value
+						}
+
+						chunk = itemSp.SetText(label + val)
+						chunk.Style.Font = font
+					}
+
+					height += itemSp.Height()
+				}
+
+				return height
+			},
+			"isFitInPageHeight": func(height float64) bool {
+				return height <= c.Height()-320
+			},
+			"add": func(a, b float64) float64 {
+				return a + b
+			},
+		},
+	}
+
+	// Draw header and footer.
+	drawHeader := func(tplPath string, block *creator.Block, version, releaseDate string, pageNum, totalPages int) {
+		// Read template.
+		tpl, err := readTemplate(tplPath)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Draw template.
+		data := map[string]interface{}{
+			"Version":     version,
+			"ReleaseDate": releaseDate,
+			"PageNum":     pageNum,
+			"TotalPages":  totalPages,
+		}
+
+		if err := block.DrawTemplate(c, tpl, data, tplOpts); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	c.DrawHeader(func(block *creator.Block, args creator.HeaderFunctionArgs) {
+		drawHeader("templates/header.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+	c.DrawFooter(func(block *creator.Block, args creator.FooterFunctionArgs) {
+		drawHeader("templates/footer.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+
+	if err := c.DrawTemplate(mainTpl, checks, tplOpts); err != nil {
```

Comment:
Please move the drawing of the main template somewhere around line `110` so that it's in the section specified by the comment: `// Draw main content template.`.